pyximport/pyximport.py

```diff
@@ -300,19 +301,31 @@ def find_module(self, fullname, package_path=None):
             paths = package_path
         else:
             paths = sys.path
-        join_path = os.path.join
-        is_file = os.path.isfile
-        is_abs = os.path.isabs
-        abspath = os.path.abspath
-        #is_dir = os.path.isdir
-        sep = os.path.sep
+
         for path in paths:
             if not path:
                 path = os.getcwd()
-            elif not is_abs(path):
-                path = abspath(path)
-            if is_file(path+sep+pyx_module_name):
-                return PyxLoader(fullname, join_path(path, pyx_module_name),
+            elif not os.path.isabs(path):
+                path = os.path.abspath(path)
+
+            try:
+                zi = zipimporter(path)
+                data = zi.get_data(pyx_module_name)
+            except (ZipImportError, IOError):
+                pyx_module_path = os.path.join(path, pyx_module_name)
```

Comment:
No, not if the shared library does not exist yet.

I just read the source and the zipimporter is actually quite efficient here. The first thing it does is to check if the path it received is a file. Still, I don't see why we shouldn't check if path points to a directory first (because that's really the only case we can handle) and only failing that, fall back to trying the zipimporter. No need to involve the zipimporter for the obvious and expected "it's a directory but it does not contain the file we search" case.